src/Sulu/Component/Content/Types/BlockContentType.php

```diff
@@ -464,4 +466,33 @@ function(ContentTypeInterface $contentType, $property) {
             false
         );
     }
+
+    public function getReferences(PropertyInterface $property, ReferenceCollectorInterface $referenceCollector, string $propertyPrefix = ''): void
+    {
+        $values = $property->getValue();
+
+        if (!\is_array($values)) {
+            return;
+        }
+
+        foreach ($values as $index => $value) {
+            $propertyType = $property->getType($value['type']);
+
+            foreach ($propertyType->getChildProperties() as $child) {
+                $contentType = $this->contentTypeManager->get($child->getContentTypeName());
+                $childName = $child->getName();
+                $oldValue = $child->getValue();
+
+                if ($contentType instanceof ReferenceContentTypeInterface) {
+                    $child->setValue($value[$childName]);
+                    $contentType->getReferences(
+                        $child,
+                        $referenceCollector,
+                        $propertyPrefix . $property->getName() . '[' . $index . '].'
+                    );
+                    $child->setValue($oldValue);
```

Comment:
@alexander-schranz let us discuss this tomorrow, I'm not sure if I'm overseeing something or if we should handle it in another way